robot/Cumulus/resources/NPSP.py

```diff
@@ -1242,7 +1243,18 @@ def randomString(self,stringLength=10):
         letters = string.ascii_lowercase
         return ''.join(random.choice(letters) for i in range(stringLength))
 
-    def setupdata(self, name, contact_data=None, opportunity_data=None, account_data=None, engagement_data=None, task_data=None):
+    def scroll_button_into_view_and_click_using_js(self, type, value):
+        xpath = self.get_npsp_locator(type,value)
+        javascript = (
+            "window.document.evaluate("
+            f"    '{xpath}', document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null"
+            ").singleNodeValue.scrollIntoView(true)"
+
+        )
+        self.selenium.execute_javascript(javascript)
```

Comment:
I have a similar case in sal repo and i had to just specify the xpath and call the below code and it works fine.
`scroll_locator=self.selenium.driver.find_element_by_xpath(locator)
 self.selenium.driver.execute_script("arguments[0].scrollIntoView(true)", scroll_locator)`
 Any reason to mention the value within the `javascript` variable?

Reply:
Yeah, I tried that way too and for some reason, this way only works on this page. I checked online and read reviews on implementing it this way.  Please let me know if there is any specific way with this implementation.

Reply:
It was only working this way, I tried many other options before concluding with this.